contracts/RewardsBooster.sol

```diff
@@ -191,21 +202,65 @@ contract RewardsBooster is Initializable, OwnableUpgradeable, IRewardsBooster {
 
     /**
      * @notice remove booster from deployment
-     * @param deployment deploymentId
+     * @param deploymentId deploymentId
      * @param amount the added amount
      */
-    function removeBoosterDeployment(bytes32 deployment, uint256 amount) external {
-        DeploymentPool storage deploymentPool = deploymentPools[deployment];
+    function removeBoosterDeployment(
+        bytes32 deploymentId,
+        uint256 amount
+    ) external {
+        DeploymentPool storage deploymentPool = deploymentPools[deploymentId];
         require(deploymentPool.accountBooster[msg.sender] >= amount, 'RB003');
 
-        onDeploymentBoosterUpdate(deployment, msg.sender);
+        onDeploymentBoosterUpdate(deploymentId, msg.sender);
         deploymentPool.boosterPoint -= amount;
         deploymentPool.accountBooster[msg.sender] -= amount;
         deploymentPool.accRewardsPerBooster = accRewardsPerBooster;
         totalBoosterPoint -= amount;
 
         IERC20(settings.getContractAddress(SQContracts.SQToken)).safeTransfer(msg.sender, amount);
-        emit DeploymentBoosterRemoved(deployment, msg.sender, amount);
+        emit DeploymentBoosterRemoved(deploymentId, msg.sender, amount);
+    }
+
+    /**
+     * @notice swap booster from one deployment to another
+     * @param account the account booster the deployments
+     * @param from from deploymentId
+     * @param to  to deploymentId
+     * @param amount the amount to swap
+     */
+    function swapBoosterDeployment(
+        address account,
+        bytes32 from,
+        bytes32 to,
+        uint256 amount
+    ) external onlyRegisteredDeployment(to) {
+        require(from != to, 'RB013');
+        if (account != msg.sender) {
+            require(
+                IConsumerRegistry(settings.getContractAddress(SQContracts.ConsumerRegistry))
+                    .isController(account, msg.sender),
+                'RB014'
+            );
+        }
+
+        DeploymentPool storage fromPool = deploymentPools[from];
+        require(fromPool.accountBooster[account] >= amount, 'RB003');
+
+        // remove booster from current deploymentId
+        onDeploymentBoosterUpdate(from, account);
```

Comment:
I prefer to have ` _removeBoosterDeployment() internal` and `_boostDeployment() internal` then reuse them